src/components/BoardView.js

```diff
@@ -17,16 +17,38 @@ const BACKGROUNDS_NR = 5;
 
 const randomBg = backgrounds[random(BACKGROUNDS_NR)];
 
+const randomCardIds = (cardCount, imgArr) => {
+  let array = shuffle(imgArr).splice(0, cardCount / 2);
+
+  const doubledArray = array.concat(array);
+
+  return shuffle(doubledArray);
+};
+
 class BoardView extends Component {
   state = {
-    cardCount: range(CARD_COUNT)
+    cardCount: []
   };
 
+  componentDidUpdate(oldProps) {
+    if (this.props !== oldProps ) {
+      this.setState({
+        cardCount: randomCardIds(CARD_COUNT, this.props.photoArray)
+      });
+    }
+  }
+
   render() {
     return (
       <div className="container">
-        {this.state.cardCount.map(card => (
-          <CardView key={card} id={card} background={randomBg} />
+        {this.state.cardCount.map((card, i) => (
+          <CardView
+            key={i}
```

Comment:
Whats `key` ? Perhaps it's redundant if it's the same as ID. I think we should introduce `/model` in which there would be objects that represent logic. Seems like BoardView should use a generator strategy to generate a number of card objects and  show them. The Card should know it works with CardView probably, not BoardView.

Reply:
[`key`](https://reactjs.org/docs/lists-and-keys.html#keys) prop is required by React in cases like this one. It allows you to distinguish components from each other.

Reply:
Regarding indexes as keys https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318